lib/ram_pwrdn/ram_pwrdn.c

```diff
@@ -4,26 +4,39 @@
  * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
  */
 
-#include <hal/nrf_power.h>
 #include <logging/log.h>
 #include <sys/util.h>
 #include <zephyr.h>
 #include <stdint.h>
 
+#if defined(CONFIG_SOC_NRF52840) || defined(CONFIG_SOC_NRF52833)
+#include <hal/nrf_power.h>
+#elif defined(CONFIG_SOC_NRF5340_CPUAPP)
+#include <hal/nrf_vmc.h>
+#else
+#error "RAM power-down module is not supported on the current platform"
+#endif
+
 #if CONFIG_RAM_POWER_ADJUST_ON_HEAP_RESIZE
 #include <init.h>
 #include <sys/heap_listener.h>
 #endif
 
-LOG_MODULE_REGISTER(ram_pwrdn, CONFIG_RAM_POWERDOWN_LOG_LEVEL);
+#if defined(CONFIG_PARTITION_MANAGER_ENABLED)
+#include <pm_config.h>
 
-#if !defined(CONFIG_SOC_NRF52840) && !defined(CONFIG_SOC_NRF52833)
-#error "RAM power-down module is not supported on the current platform"
+#define RAM_APP_END_ADDR_PM ((uintptr_t)PM_SRAM_PRIMARY_END_ADDRESS)
+#define RAM_APP_END_ADDR_DTS ((uintptr_t)(CONFIG_SRAM_SIZE * 1024UL + CONFIG_SRAM_BASE_ADDRESS))
```

Comment:
Is there any use case for DTS having a smaller RAM partition than the partition manager? Shouldn't we rather completely rely on the partition manager if it's enabled?

Reply:
We should - this check against DTS was added because of missing partition entry for rpmsg shared memory which caused problems with initial nRF53 tests. It will be fixed soon by https://github.com/nrfconnect/sdk-nrf/pull/6672
I will remove this DTS check - it has never been used for nRF52 and multi-image builds use partition manager anyway